contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -77,10 +83,45 @@ func createReceiverFunction(s *SubscriptionsSupervisor, logger *zap.SugaredLogge
 }
 
 func (s *SubscriptionsSupervisor) Start(stopCh <-chan struct{}) error {
+	// Trigger Connect to establish connection with NATS
+	s.connect <- struct{}{}
 	s.receiver.Start(stopCh)
 	return nil
 }
 
+func (s *SubscriptionsSupervisor) connectWithRetry() {
+	// re-attempting evey 60 seconds until the connection is established.
+	ticker := time.NewTicker(60 * time.Second)
+	defer ticker.Stop()
+	for {
+		nConn, err := stanutil.Connect(clusterchannelprovisioner.ClusterId, clientID, s.natssURL, s.logger.Sugar())
+		if err == nil {
+			s.subscriptionsMux.Lock()
```

Comment:
Is this now intended to lock `s.natssConn` as well?

Reply:
Right, adding a separate mutex just for natssConn does not seem to be efficient.

Reply:
My preference would be to have a second mux, `natssConnMux` for natssConn and natssConnInProgress.